### PR TITLE
Update port of all of TriBITS to Python 3.5.2 (#309)

### PR DESCRIPTION
**Pull request by bartlettroscoe**
This complete the port of all of the tested Python code in TriBITS to Python 3.5.2 and sets up the pre-push testing process to maintain the porting to Python 3.5.2 and Python 2.7.5.

### CODE REVIEW

**Comment by jmgate** [comment]
Thanks for doing this, @bartlettroscoe.

File: test/ci_support/clone_extra_repos_UnitTests.py [hunk 1]
```diff
@@ -346,7 +345,7 @@ def test_clone_git_repos_already_exists(self):
         "\n" \
         "Cloning repo extraTrilinosRepo ...\n" \
         "\n" \
-        "  ==> Repo dir = 'extraTrilinosRepo' already exists.  Skipping clone!\n")
+        "  ==> Repo dir = 'extraTrilinosRepo' already exists.  Skipping clone!\n"
```
Comment (jmgate) [inline test/ci_support/clone_extra_repos_UnitTests.py:348]: Did you accidentally lose a closing parenthesis here?

Comment (jmgate) [reply test/ci_support/clone_extra_repos_UnitTests.py:348]: Never mind.

File: tribits/doc/developers_guide/TribitsDevelopersGuide.rst [hunk 2]
```diff
@@ -6731,20 +6731,20 @@ included in the tarball.
...
+*you add in an exclude like ``"*.pyc"`` (i.e. trying to exclude all of the
+*generated Python byte code files) that it will exclude every file that has
+*``"pyc"`` in the name and **not** just those with the file extension
+*``"pyc"``.  For example, the exclude ``".pyc"`` would exclude the files
+*``"puppyc"``, ``"lpycso"``, etc.  If you want to exclude all files with
+*extension ``"pyc"``, you have to add the exclude regex ``".*[.]pyc$"``!
+*One's lack of understanding of this fact will cost someone hours of lost time
+*debugging what happens when random files are missing when one tries to
+*configure what is left.  Somethings, what is left will actually configure and
+*might almost build!
```
Comment (jmgate) [inline tribits/doc/developers_guide/TribitsDevelopersGuide.rst:6747]: Are these lines supposed to have the initial asterisk?

**Review by jmgate — COMMENTED** [review]
This looks great.  Only one question, but not a crucial one.

File: tribits/doc/developers_guide/TribitsDevelopersGuide.rst (hunk 2 above)
Comment (bartlettroscoe) [reply tribits/doc/developers_guide/TribitsDevelopersGuide.rst:6747]: No, they are not.  Thanks for the catch.  I will fix this and post to a new PR.

**Pull request merged**